alvr/adb/src/lib.rs

```diff
@@ -24,7 +25,7 @@ impl WiredConnection {
         Ok(Self { adb_path })
     }
 
-    pub fn setup(&self, control_port: u16, stream_port: u16) -> Result<WiredConnectionStatus> {
+    pub fn setup(&self, control_port: u16, settings: &Settings) -> Result<WiredConnectionStatus> {
```

Comment:
I would pass the minimum setting subset of settings as parameter. This would be `ConnectionConfig`